contracts/DomainRegistry.sol

```diff
@@ -46,28 +65,37 @@ contract DomainRegistry {
         _;
     }
 
-    constructor() {
-        owner = payable(msg.sender);
-    }
-
     function getDomain(string memory domainName) public view returns (address) {
         string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
 
         return domains[rootDomain];
     }
 
-    function registerDomain(string memory domainName) external payable checkDomainLength(domainName) checkDomainParent(domainName) checkDomainAvailability(domainName) checkSufficientEtn()
+    function registerDomain(string memory domainName) external payable
+        checkDomainLength(domainName)
+        checkDomainParent(domainName)
+        checkDomainAvailability(domainName)
+        checkSufficientEtn(domainName)
     {
         string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
+        uint256 price = getDomainPrice(domainName);
```

Comment:
What if you forgot to set the domain price for some level? The domain will be free. Is that your intention?
Except that, there is some small chance that someone can jump in between the deployment and your setDomainPrice() transaction to take some domains for free.

Reply:
added setup of default prices in constructor https://github.com/vpmoon/blockchain-playground/pull/5/commits/00b1a5dac7cb5e778be1f20cb9d1d75c3a2856ef